source-control/git-guidance/readme.md

```diff
@@ -0,0 +1,168 @@
+# Git Guidance
+
+## The basics
+
+### Installation
+
+[Install Git](https://git-scm.com/downloads) and follow the [First-Time Git Setup](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup).
+
+### Cloning
+
+<!-- TODO: add more information about cloning repos (e.g., SSH, https, etc.) -->
+
+### Committing
+
+<!-- TODO: add more information about commiting changes (e.g., commit messages, setting editors, etc.) -->
+
+### Merging
+
+<!-- TODO: add more information about merging changes (e.g., links) -->
+
+### Managing remotes
+
+<!-- TODO: add more information about managing remotes (e.g., origin, other GitHub forks, etc., SSH vs. HTTPS) -->
+
+## Rolling back changes
+
+### Reverting and deleting commits
+
+To "undo" a commit, run the following two commands: `git revert` and `git reset`. `git revert` creates a new commit that undoes commits while `git reset` allows to delete commits entirely from the commit history.
+
+> If you have committed secrets/keys, `git reset` will remove them from the commit history!
+
+To **delete** the latest commit use `HEAD~`:
+
+```bash
+git reset --hard HEAD~1
+```
+
+To delete commits back to a specific commit, use the respective commit id:
+
+```bash
+git reset --hard <sha1-commit-id>
+```
+
+after you deleted the unwanted commits, push using `force`:
+
+```bash
+git push origin HEAD --force
+```
+
+<!-- TODO: add information about interactive rebase -->
+
+### Using the reflog
+
+<!-- TODO: add information about using the reflog to recover "lost" changes -->
+
+## Using submodules
```

Comment:
This is a completely new section added as part of a restructure.  Following our own guidelines above this should also not be added in this PR.  I suggest that we remove this and add an issue with this content to be added after this restructure

## Commit Best Practices

* Make small commits. This makes changes easier to review, and if we need to revert a commit, we lose less work.
* Commit complete and well tested code. Never commit incomplete code, get in the habit of testing your code before committing.
* Don't mix whitespace changes with functional code changes. It is hard to determine if the line has a functional change or only removes a whitespace, so functional changes may go unnoticed.
* Write good commit messages.

Reply:
I don't believe I wrote this. I think I moved it in from the main README or something. Will double check.

Reply:
This was moved from the main README to the contributing README. Prefer to leave as is.

Reply:
I finally found where it came from... :) it was a module in the git.md originally named Multi service deployment and deployments - this was why I didn't find it